mods/creative/inventory.lua

```diff
@@ -0,0 +1,162 @@
+creative = {}
+local player_inventory = {}
+
+function creative.init_creative_inventory(player)
+	local player_name = player:get_player_name()
+	player_inventory[player_name] = {
+		size = 0,
+		filter = "",
+		start_i = 1
+	}
+
+	minetest.create_detached_inventory("creative_" .. player_name, {
+		allow_move = function(inv, from_list, from_index, to_list, to_index, count, player2)
+			if not to_list == "main" then
+				return count
+			else
+				return 0
+			end
+		end,
+		allow_put = function(inv, listname, index, stack, player2)
+			return 0
+		end,
+		allow_take = function(inv, listname, index, stack, player2)
+			return -1
+		end,
+		on_move = function(inv, from_list, from_index, to_list, to_index, count, player2)
+		end,
+		on_put = function(inv, listname, index, stack, player2)
+		end,
+		on_take = function(inv, listname, index, stack, player2)
+			if stack and stack:get_count() > 0 then
+				minetest.log("action", player_name .. " takes " .. stack:get_name().. " from creative inventory")
+			end
+		end,
+	})
+
+	creative.update_creative_inventory(player_name, minetest.registered_items)
+end
+
+function creative.update_creative_inventory(player_name, tab_content)
+	local creative_list = {}
+	local player_inv = minetest.get_inventory({type = "detached", name = "creative_" .. player_name})
+	local inv = player_inventory[player_name]
+
+	for name, def in pairs(tab_content) do
+		if not (def.groups.not_in_creative_inventory == 1) and
```

Comment:
this certainly needs to be discussed. I'm not entirely sure what groups are meant to be